projects/components/src/table/controls/table-controls.component.ts

```diff
@@ -0,0 +1,114 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
+import { SubscriptionLifecycle, TypedSimpleChanges } from '@hypertrace/common';
+import { Subject } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+import { ToggleItem } from '../../toggle-group/toggle-item';
+import { TableMode } from '../table-api';
+
+@Component({
+  selector: 'ht-table-controls',
+  styleUrls: ['./table-controls.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="table-controls">
+      <!-- Filters -->
+      <div class="filter-controls">
+        <!-- Search -->
+        <ht-search-box
+          *ngIf="this.searchAttribute"
+          class="control search-box"
+          [placeholder]="this.searchPlaceholder"
+          (valueChange)="this.onSearchChange($event)"
+        ></ht-search-box>
+
+        <!-- Filter Toggle -->
+        <ht-toggle-group
+          *ngIf="this.filterItems?.length"
+          class="control filter-toggle-group"
+          [items]="this.filterItems"
+          [activeItem]="this.activeFilterItem"
+          (activeItemChange)="this.onFilterChange($event)"
+        ></ht-toggle-group>
+      </div>
+
+      <!-- Mode Toggle -->
+      <ht-toggle-group
+        *ngIf="this.modeItems?.length"
+        class="control mode-toggle-group"
+        [items]="this.modeItems"
+        [activeItem]="this.activeModeItem"
+        (activeItemChange)="this.onModeChange($event)"
+      ></ht-toggle-group>
+    </div>
+  `
+})
+export class TableControlsComponent implements OnChanges {
+  @Input()
+  public searchAttribute?: string;
+
+  @Input()
+  public searchPlaceholder?: string = 'Search...';
+
+  @Input()
+  public filterItems?: ToggleItem[] = [];
+
+  @Input()
+  public activeFilterItem?: ToggleItem;
+
+  @Input()
+  public modeItems?: ToggleItem[] = [];
+
+  @Input()
+  public activeModeItem?: ToggleItem;
+
+  @Output()
+  public readonly searchChange: EventEmitter<string> = new EventEmitter<string>();
+
+  @Output()
+  public readonly filterChange: EventEmitter<ToggleItem> = new EventEmitter<ToggleItem>();
+
+  @Output()
+  public readonly modeChange: EventEmitter<TableMode> = new EventEmitter<TableMode>();
+
+  private readonly searchDebounceSubject: Subject<string> = new Subject<string>();
+
+  public constructor(private readonly subscriptionLifecycle: SubscriptionLifecycle) {
+    this.subscriptionLifecycle.add(
+      this.searchDebounceSubject.pipe(debounceTime(200)).subscribe(text => this.searchChange.emit(text))
```

Comment:
note to self - add this support natively to searchbox

Reply:
Yeah, the way I'd expect this to work would be an input like `debounceTime` that defaults to 0, so any debounce can be specified and it's backwards compat.